lib/_http_server.js

```diff
@@ -273,7 +277,9 @@ function Server(requestListener) {
   // http://wiki.squid-cache.org/SquidFaq/InnerWorkings#What_is_a_half-closed_filedescriptor.3F
   this.httpAllowHalfOpen = false;
 
-  this.on('connection', connectionListener);
+  this.on('connection', (socket) => defaultTriggerAsyncIdScope(
```

Comment:
Since this is hot path listener, this should be refactored to not have a closure. The fact that an Array is now created each time is also problematic.

Could we perhaps instead keep all the `connectionListener` references as is, then rename `connectionListener` to something like `internalConnectionListener` and have the `connectionListener` invoke the `defaultTriggerAsyncIdScope` with `internalConnectionListener`? That way there's no need to make changes in `https` or `http2`.

(Also, I'll open a PR shortly to update `defaultTriggerAsyncIdScope` for performance.)

Reply:
This is what I had in mind: https://github.com/nodejs/node/commit/a530e8f7730cd648aaf5528cbea63a84c35ff3ef — you should be able to apply it as a patch if you want.

Reply:
Hehe, sorry. Already done it locally, running tests ...

Reply:
No worries. Just put it together in case I wasn't being clear and/or you didn't have time to work on it :)